processes-endpoint.md

```diff
@@ -0,0 +1,215 @@
+        "parameters" : [
+          {
+            "name" : "c",
+            "value" : "954e5cfa-6990-4c85-ae42-f30d8c7888e2"
+          },
+          {
+            "name" : "n",
+            "value" : "true"
+          }
+        ],
+        "_links" : {
+          "self" : {
+            "href" : "/api/system/processes/000003f1-a850-49de-af03-997272d834c9"
+          },
+          "script" : {
+            "href" : "/api/system/scripts/import"
+          },
+          "output" : {
+            "href" : "/api/system/processes/000003f1-a850-49de-af03-997272d834c9/output"
+          },
+          "files" : {
+            "href" : "/api/system/processes/000003f1-a850-49de-af03-997272d834c9/files"
+          }
+        }
+      },
+      {
+        "processId" : "d3007ea8-1cc0-481d-914a-06720a200edf",
+        "userId" : "c7d85e7f-63e5-4bc0-96cb-5d80be48d62e",
+        "startTime" : "2017-11-20T10:29:11Z",
+        "status" : "FAILED",
+        "parameters" : [
+          {
+            "name" : "i",
+            "value" : "c70893a6-ac55-48c7-9447-61e026b62929"
+          }
+        ],
+        "_links" : {
+          "self" : {
+            "href" : "/api/system/processes/000003f1-a850-49de-af03-997272d834c9"
+          },
+          "script" : {
+            "href" : "/api/system/scripts/metadata-export"
+          },
+          "output" : {
+            "href" : "/api/system/processes/000003f1-a850-49de-af03-997272d834c9/output"
+          },
+          "files" : {
+            "href" : "/api/system/processes/000003f1-a850-49de-af03-997272d834c9/files"
+          }
+        }
+      }
+    ]
+  }
+}
+```
+
+Optional parameters to query the processes:
+* `script`: The name of the script (e.g. import)
+* `userId`: The UUID of the person who created the process (only admins can retrieve processes they didn't start)
+* `status`: The status of the script
+* `parameter.xyz`: Which parameters have been used, and their value
+
+Sort options should support:
+* `script`
+* `startTime`
+
+## Execution Details
+**GET /api/system/processes/<:process-id>**
+
+This endpoint will return details on the requested process.
+
+```json
+{
+  "processId" : "000003f1-a850-49de-af03-997272d834c9",
+  "userId" : "aa0263e2-b90a-4528-89fa-116ea4859de1",
+  "startTime" : "2017-11-22T10:29:11Z",
+  "status" : "RUNNING",
+  "parameters" : [
+    {
+    "name" : "c",
+    "value" : "954e5cfa-6990-4c85-ae42-f30d8c7888e2"
+    },
+    {
+    "name" : "n",
+    "value" : "true"
+    }
+  ],
+  "_links" : {
+    "self" : {
+      "href" : "/api/system/processes/000003f1-a850-49de-af03-997272d834c9"
+    },
+    "script" : {
+      "href" : "/api/system/scripts/import"
+    },
+    "output" : {
+      "href" : "/api/system/processes/000003f1-a850-49de-af03-997272d834c9/output"
+    },
+    "files" : {
+      "href" : "/api/system/processes/000003f1-a850-49de-af03-997272d834c9/files"
+    }
+  }
+}
+```
+
+The possible `status` values are `RUNNING`, `STOPPED` and `FAILED`.
```

Comment:
is COMPLETED a better choice than STOPPED? STOPPED or KILLED should be reserved for the future support of DELETE